### PR TITLE
Custom URL Admin API implementation

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
This API is implemented to fetch custom URL information to Analytics Server.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/CustomUrlApi.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.wso2.carbon.apimgt.rest.api.admin;
```
Comment (TanyaM) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/CustomUrlApi.java]: License missing

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml [hunk 2]
```diff
@@ -2626,6 +2626,57 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 
+  ######################################################
+  # The Custom URL Info resource API
+  ######################################################
+  /custom-url/{tenantDomain}:
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]:
```suggestion
/custom-urls/{tenantDomain}:
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/CustomUrlApiService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package org.wso2.carbon.apimgt.rest.api.admin;
```
Comment (TanyaM) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/CustomUrlApiService.java]: license?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml [hunk 4]
```diff
@@ -3779,6 +3830,27 @@ definitions:
...
+    title: Custom url information
+    type: object
+    description: The custom url information of the tenant domain
+    properties:
+      tenantDomain:
+        type: string
+        example: 'carbon.super'
+      tenantAdminUsername:
+        type: string
+        example: 'john@foo.com'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]: Are these required?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml [hunk 5]
```diff
@@ -3779,6 +3830,27 @@ definitions:
...
+      isCustomUrlEnabled:
+        type: boolean
+        example: true
+      customUrl:
+        type: string
+        example: 'https://example.org/analytics'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]: customUrls:
        properties:
          devportal:
            type: string
            example: 'https://example.org/devportal'
          analytics:
            type: string
            example: 'https://example.org/analytics'

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml (hunk 4 above)
Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]: The response schema is expected to have these keys.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml (hunk 5 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]: How about below:
```
      customUrls:
        properties:
          devportal:
            type: string
            example: 'https://example.org/devportal'
          analytics:
            type: string
            example: 'https://example.org/analytics'
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml [hunk 6]
```diff
@@ -3779,6 +3830,27 @@ definitions:
...
+    type: object
+    description: The custom url information of the tenant domain
+    properties:
+      tenantDomain:
+        type: string
+        example: 'carbon.super'
+      tenantAdminUsername:
+        type: string
+        example: 'john@foo.com'
+      isCustomUrlEnabled:
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]: enabled: true/false

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml (hunk 5 above)
Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml]: changed as suggested

**Pull request merged**